lib/cdo/rack/whitelist.rb

```diff
@@ -0,0 +1,107 @@
+# Rack middleware that whitelists cookies and headers based on path-based cache behaviors.
+# Behaviors are defined in http cache config.
+require_relative '../../../cookbooks/cdo-varnish/libraries/helpers'
+require 'active_support/core_ext/hash/slice'
+require 'cdo/rack/response'
+require 'cdo/aws/cloudfront'
+
+module Rack
+  module Whitelist
+    # Downstream middleware filters out unwanted HTTP request headers and cookies,
+    # and extracts cookies into HTTP headers before the request reaches the cache.
+    class Downstream
+      attr_reader :config
+
+      def initialize(app, config)
+        @app = app
+        @config = config
+      end
+
+      def call(env)
+        return [403, {}, ['Unsupported method.']] unless AWS::CloudFront::ALLOWED_METHODS.include?(env['REQUEST_METHOD'].upcase)
+        request = Rack::Request.new(env)
+        path     = request.path
```

Comment:
You don't have to line up the equals, or if you do please line up with line 22 as well.